sddi/Dockerfile

```diff
@@ -1,107 +1,193 @@
+  pip install --no-index --find-links=${APP_DIR}/ext_wheels ckanext-security
+
+ENV CKAN__PLUGINS "envvars image_view text_view webpage_view datastore \
+  tracking \
+  activity \
+  security \
+  hierarchy_display hierarchy_form \
+  relation \
+  spatial_metadata spatial_query \
+  datesearch \
+  scheme_sddi \
+  theme_sddi \
+  scheming_datasets \
+  geo_view geojson_view wmts_view shp_view \
+  heroslideradmin \
+  dcat dcat_json_interface \
+  clamav"
+
+RUN set -ex && \
+  ckan generate config ${APP_DIR}/production.ini
+
 RUN set -ex && \
   ckan config-tool "${CKAN_INI}" "ckan.plugins = ${CKAN__PLUGINS}" && \
-  ckan config-tool "${CKAN_INI}" "ckanext.geoview.ol_viewer.formats = wms kml" && \
-  ckan config-tool "${CKAN_INI}" "ckanext.geoview.shp_viewer.srid = 4326" && \
-  ckan config-tool "${CKAN_INI}" "ckanext.geoview.shp_viewer.encoding = UTF-8" && \
-  ckan config-tool "${CKAN_INI}" "who.timeout = 1800" && \
-  ckan config-tool "${CKAN_INI}" "ckan.auth.public_user_details = False" && \
-  # Remove wheels
+  ckan config-tool "${CKAN_INI}" "ckan.spatial.srid = 4326" && \
+  ckan config-tool "${CKAN_INI}" "ckanext.spatial.search_backend = solr-bbox" && \
+  ckan config-tool "${CKAN_INI}" "ckanext.spatial.use_postgis_sorting = true" && \
+  ckan config-tool "${CKAN_INI}" "scheming.dataset_schemas = ckanext.scheme_sddi:sddi_dataset.yaml" && \
+  ckan config-tool "${CKAN_INI}" "scheming.presets = ckanext.scheming:presets.json ckanext.scheme_sddi:sddi_presets.json" && \
+  ckan config-tool "${CKAN_INI}" "scheming.dataset_fallback = false" && \
+  ckan config-tool "${CKAN_INI}" "ckanext.dathere_theme.column_count = 4" && \
+  ckan config-tool "${CKAN_INI}" "ckanext.spatial.common_map.type = custom" && \
+  ckan config-tool "${CKAN_INI}" "ckanext.spatial.common_map.custom.url = https://tile.openstreetmap.de/{z}/{x}/{y}.png" && \
+  ckan config-tool "${CKAN_INI}" "ckanext.spatial.common_map.attribution = <a href="https://www.openstreetmap.org/copyright">OpenStreetMap</a> contributors." && \
+  ckan config-tool "${CKAN_INI}" "ckanext.security.lock_timeout = 900" && \
+  ckan config-tool "${CKAN_INI}" "ckanext.security.login_max_count = 3" && \
+  ckan config-tool "${CKAN_INI}" "ckanext.security.brute_force_key = user_name" && \
+  ckan config-tool "${CKAN_INI}" "ckanext.security.disable_password_reset_override = true" && \
+  ckan config-tool "${CKAN_INI}" "ckanext.security.enable_totp = true" && \
+  ckan config-tool "${CKAN_INI}" "ckanext.dcat.enable_content_negotiation = True" && \
+  ckan config-tool "${CKAN_INI}" "PERMANENT_SESSION_LIFETIME = 600"
+
+# Remove wheels
+RUN set -ex && \
   rm -rf ${APP_DIR}/ext_wheels
 
-USER ckan
+WORKDIR ${APP_DIR}
+
+# Create entrypoint directory for children image scripts
+ONBUILD RUN mkdir -p /docker-entrypoint.d
+
+EXPOSE 5000
+HEALTHCHECK --interval=60s --timeout=5s --retries=5 CMD curl --fail http://localhost:5000/api/3/action/status_show || exit CMD ["/srv/app/start_ckan.sh"]
```

Comment:
This won't work if somebody changes the CKAN port. Is there a variable you can use here?

Reply:
CKAN has used port 5000 since 2006, if somebody changes the port I am sure that many other things will be changed and we will need to adapt to all of them somehow.

The port is hardcoded in the base image as well https://github.com/ckan/ckan-docker-base/blob/main/ckan-2.11/Dockerfile#L108.